specifyweb/frontend/js_src/lib/components/FormSliders/AttachmentsCollection.tsx

```diff
@@ -44,9 +44,19 @@ export function AttachmentsCollection({
     })
   );
 
+  const [isAttachmmentsSaved, setIsAttachmmentsSaved] = React.useState(false);
+
+  React.useEffect(() => {
+    const savedAttachments = attachments.filter(
+      (attachment) => attachment.attachmentLocation !== null
+    );
+    setIsAttachmmentsSaved(savedAttachments.length === attachments.length);
+  }, attachments);
```

Comment:
(line 54) this looks like a bug.
while yes attachments is an array, and useEffect (or useMemo) accept an array, there are two problems here:
1. This will make react check every attachment in the array to see if it changed - that is redundant because in react normally on any change to item in an array, entire array is changed, so it's sufficient to just check if array is the same - no need to check individual array items (that's the benefit of not mutating things)
2. The length of the dependency array should not change - if it does, react doesn't like it and can emit warnings

thus, you should use `[attachments]` instead of `attachments`

Reply:
but, I also just noticed that `attachments` is an array that is created from scratch on each render - so it will cause useEffect to be called every time - no point in useEffecy/useMemo then